thinc/loss.py

```diff
@@ -101,6 +104,9 @@ def convert_truths(self, truths, guesses: Floats2d) -> Tuple[Floats2d, Floats2d]
         if truths.ndim != guesses.ndim:
             # transform categorical values to one-hot encoding
             truths = to_categorical(cast(Ints1d, truths), n_classes=guesses.shape[-1])
+            # Only smooth one-hot encoding
+            if self.alpha > 0:
```

Comment:
We should probably throw an exception if label smoothing was enabled but not applied (e.g. because the user provided a truth matrix).

Reply:
I think raising an exception would be the best, since this would be a programming error (use of incompatible arguments).